app/models/mixins/maintenance_mixin.rb

```diff
@@ -0,0 +1,17 @@
+module MaintenanceMixin
+  extend ActiveSupport::Concern
+
+  class_methods do
+    def reindex
+      _log.info("Reindexing table #{reindex_table_name}")
+      result = ActiveRecord::Base.connection.execute("REINDEX TABLE #{reindex_table_name}")
```

Comment:
We can use the methods defined [here](https://github.com/ManageIQ/manageiq/blob/master/lib/extensions/ar_adapter/ar_dba.rb#L861) for this.